src/print.rs

```diff
@@ -450,6 +452,57 @@ pub fn print_schema() {
     println!("{}", serde_json::to_string_pretty(&schema).unwrap());
 }
 
+#[derive(clap::ValueEnum, Clone, Debug)]
+pub enum Preset {
+    BracketedSegments,
+    NerdFontSymbols,
+    NoRuntimeVersions,
+    PastelPowerline,
+    PlainTextSymbols,
+    PurePreset,
+}
```

Comment:
I actually meant handling the validation via a type that manually implement the `clap::ValueEnum` trait, by looking at the values in `Presets`, instead of the manual validation you had before.
If the implementation does require presets to be added to the code manually, please don't use `rust-embed`.

Reply:
I am quite new to Rust so there might be another way around. The only way I can think of implementing `fn value_variants<'a>() -> &'a [Self]` method without explicitly listing all the options would be to create a procedural macro. As far as I understand, proc macros require a separate crate to maintain, which would be an unreasonable thing to do for this small feature.

I liked the idea of not listing presets manually, but if my understanding is correct, I will rewrite it using `include_bytes!`.

Reply:
I don't think it needs to be an enum, `struct Preset(String)` should work without macros.

Reply:
I've tried implementing it this way, the problem that I have is a lifetime of a slice. If I create it's value in runtime inside `value_variants`, then it's lifetime is limited to this function and thus I can't return it. So it has to be a compile time constant

Reply:
Could `struct Preset(&'static str)` work?

Reply:
That solves part of the problem. The other part is from where `&[Preset]` will be borrowed from. The only thing I can think of is using something like `static mut PRESETS: Vec<Preset> = Vec::new() `, changing it in the method and returning it's slice, but that requires `unsafe` block and overall feels ugly